README.md

```diff
@@ -19,11 +19,72 @@ Install ruby and set your local ruby version to `2.6.5`
 The main template is `rails_docker`. In order to use it, initialize a new app with the following parameters:
 
 ```
-rails new <app_name> -m https://raw.githubusercontent.com/nimblehq/rails-templates/master/template.rb -T
+rails new <app_name> -m https://raw.githubusercontent.com/nimblehq/rails-templates/master/template.rb
 ```
 
+Supported template options:
+- `--api` - create an api-only application
+
 Read more about Rails Application Template in the [official Rails Guides](https://guides.rubyonrails.org/rails_application_templates.html).
 
+## How to contribute
+
+#### Template structure
+
+```
+.
+├── .template
+│   ├── addons
+│   │   └── docker
+│   │       ├── ...
+│   │       └── template.rb
+│   └── variants
+│       ├── api
+│       │   ├── ...
+│       │   └── template.rb
+│       └── web
+│           ├── ...
+│           └── template.rb
+├── app
+├── bin
+├── config
+├── spec
+├── ...
+├── README.md
+├── README.md.tt
+└── template.rb
+```
+
+We keep the Rails-app-like structure. On the root, there are base project file templates. 
+Other files including the template options are in `.template` folder.
+
+There are 2 kinds of the template options:
+
+1. **Variants** - For the app main options, which is `web` and `api`.
+    
+2. **Addons** - For other extra options that we can add to the project like `docker` or `bootstrap`. 
+The questions will be prompted before generating the project.
```

Comment:
This needs to be in one sentence:

```
For other extra options that we can add to the project like `docker` or `bootstrap`, use the prompt `ask` to generate a question before generating the project.
```

Also technically this is no implemented yet. But can we put an example of how it would look like and also where it should find in the root template file?